low-level/files/opt/seagate/sspl/setup/cortx_manifest/lr2/manifest_storage.py

```diff
@@ -0,0 +1,381 @@
+                f"PSU Manifest Data:{data}"))
+        return data
+
+    def get_drives_info(self):
+        """Update and return drives information in specific format"""
+        data = []
+        drives = self.get_realstor_encl_data("disks")
+        for drive in drives:
+            drive_dict = {
+                "uid": drive.get("durable-id", "NA"),
+                "type": drive.get("type", "NA"),
+                "description": drive.get("description", "NA"),
+                "product": drive.get("object-name", "NA"),
+                "manufacturer": drive.get("vendor", "NA"),
+                "serial_number": drive.get("serial-number", "NA"),
+                "version": drive.get("hardware-version", "NA"),
+                "part_number": drive.get("part-number", "NA"),
+                "last_updated": int(time.time()),
+                "specifics": [{
+                    "drive-serial-number": drive.get("serial-number")[:8],
+                    "model": drive.get("model", "NA"),
+                    "slot": drive.get("slot", "NA"),
+                    "architecture": drive.get("architecture", "NA"),
+                    "interface": drive.get("interface", "NA"),
+                    "usage": drive.get("usage", "NA"),
+                    "current_job_completion": drive.get("current-job-completion", "NA"),
+                    "speed": drive.get("speed", "NA"),
+                    "size": drive.get("size", "NA"),
+                    "enclosure_wwn": drive.get("enclosure-wwn", "NA"),
+                    "status": drive.get("status", "NA"),
+                    "ssd_life_left": drive.get("ssd-life-left", "NA"),
+                    "led_status": drive.get("led-status", "NA"),
+                    "temperature": drive.get("temperature", "NA"),
+                    "location": drive.get("location", "NA")
+                }]
+            }
+            data.append(drive_dict)
+            logger.debug(self.log.svc_log(
+                f"Drive Manifest Data:{data}"))
+        return data
+
+    @staticmethod
+    def get_fan_specfics(fan):
+        return {
+            "uid": fan.get("durable-id", "NA"),
+            "type": fan.get("type", "NA"),
+            "description": fan.get("description", "NA"),
+            "product": fan.get("object-name", "NA"),
+            "manufacturer": fan.get("vendor", "NA"),
+            "serial_number": fan.get("serial-number", "NA"),
+            "version": fan.get("hardware-version", "NA"),
+            "part_number": fan.get("part-number", "NA"),
+            "specifics":[{
+                "status": fan.get("status", "NA"),
+                "speed": fan.get("speed", "NA"),
+                "location": fan.get("location", "NA")
+            }]
+        }
+
+    def get_fan_modules_info(self):
+        """Update and return fan modules information in specific format"""
+        data = []
+        fanmoduels_data = self.get_realstor_encl_data("fan-modules")
+        if fanmoduels_data:
```

Comment:
In all functions, log the error message if `get_realstor_encl_data` is not returning anything.

Reply:
done